### PR TITLE
Update Application Config Docs

### PR DESCRIPTION
**Pull request by wascloud**
### Outline

1. Update docs for application config to use __APP_CONFIG__
2. Add .idea to .gitignore

###### Check List

- [x] commits are squashed 
- [x] commits follow the [Karma format][karma-format]

[karma-format]: karma-runner.github.io/1.0/dev/git-commit-msg.html

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/.idea
```
Comment (ahmadnassri) [inline .gitignore]: why is this here?

use your global `.gitignore` file

File: development/application-configuration.md [hunk 2]
```diff
@@ -28,7 +28,20 @@ Usually the shared libraries we use also take in their configuration as JavaScri
...
+In our isomorphic [starter kit](starter-kits.md), we want to support per-environment configuration files on the client side, without having to re-run [webpack compilation](webpack.md#how). At the same time, we don't want our bundle to contain every possible environment configuration (especially in production). To accomplish this we inject the environment specific configuration into the client-side JavaScript (as `window.__APP_CONFIG__`) during isomorphic render. Throughout the codebase the config that is included looks like this:
+
+```
+if (process.env.BROWSER) {
+ envConfig = window.__APP_CONFIG__;
+} else {
+ // This will only happen on the server side
+ const env = process.env.APP_ENV || 'production';
```
Comment (tavogel) [inline development/application-configuration.md]: default APP_ENV value is development

File: development/application-configuration.md [hunk 3]
```diff
@@ -28,7 +28,20 @@ Usually the shared libraries we use also take in their configuration as JavaScri
 
 #### Isomorphic APP_ENV
 
-In our isomorphic [starter kit](starter-kits.md), we want to support per-environment configuration files on the client side, without having to re-run [webpack compilation](webpack.md#how). As such, we inject APP_ENV into the client-side JavaScript (as `window.__APP_ENV__`). This should be used to load the relevant configuration file on the client side (v.s. `process.env.APP_ENV` on server-side). Use `process.env.BROWSER` to determine whether you are on client side, or not, and choose the correct APP_ENV source accordingly (note: `process.env.APP_ENV` is not available on client-side.
+In our isomorphic [starter kit](starter-kits.md), we want to support per-environment configuration files on the client side, without having to re-run [webpack compilation](webpack.md#how). At the same time, we don't want our bundle to contain every possible environment configuration (especially in production). To accomplish this we inject the environment specific configuration into the client-side JavaScript (as `window.__APP_CONFIG__`) during isomorphic render. Throughout the codebase the config that is included looks like this:
+
+```
```
Comment (darcyclarke) [inline development/application-configuration.md]: If you specify `js` and/or `javascript` you'll get syntax highlighting. 

ie.
```js
if ( this ) {
  well.then(that)
}
```

**Comment by ahmadnassri** [comment]
@wascloud poke
